src/capi.rs

```diff
@@ -383,8 +383,8 @@ unsafe fn option_match(
     "threads" => (*cfg).cfg.threads = value.parse().map_err(|_| ())?,
 
     "tiles" => enc.tiles = value.parse().map_err(|_| ())?,
-    "tile_rows_log2" => enc.tile_rows_log2 = value.parse().map_err(|_| ())?,
-    "tile_cols_log2" => enc.tile_cols_log2 = value.parse().map_err(|_| ())?,
+    "tile_rows" => enc.tile_rows = value.parse().map_err(|_| ())?,
+    "tile_cols" => enc.tile_cols = value.parse().map_err(|_| ())?,
```

Comment:
Is it preferrable to fail during parsing?

Will the `new_context` error info be properly propagated/surfaced to the user (i.e. they know what failed)?

Reply:
Probably it would be a good idea to fail on setting in the `capi`.

Reply:
Sure, I can do that.